app/src/main/java/org/thoughtcrime/securesms/util/ConversationShortcutPhoto.java

```diff
@@ -0,0 +1,217 @@
+    messageDigest.update(avatarObject.getBytes());
+    messageDigest.update(isSystemContactPhoto() ? (byte) 1 : (byte) 0);
+    messageDigest.update(ByteUtil.longToByteArray(getFileLastModified()));
+    messageDigest.update(ByteUtil.longToByteArray(VERSION));
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    ConversationShortcutPhoto that = (ConversationShortcutPhoto) o;
+    return Objects.equals(recipient, that.recipient)             &&
+           Objects.equals(avatarObject, that.avatarObject)       &&
+           isSystemContactPhoto() == that.isSystemContactPhoto() &&
+           getFileLastModified() == that.getFileLastModified();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(recipient, avatarObject, isSystemContactPhoto(), getFileLastModified());
+  }
+
+  private boolean isSystemContactPhoto() {
+    return recipient.getContactPhoto() instanceof SystemContactPhoto;
+  }
+
+  private long getFileLastModified() {
+    if (!recipient.isSelf()) {
+      return 0;
+    }
+
+    return AvatarHelper.getLastModified(ApplicationDependencies.getApplication(), recipient.getId());
+  }
+
+  public static final class Loader implements ModelLoader<ConversationShortcutPhoto, Bitmap> {
+
+    private final Context context;
+
+    private Loader(@NonNull Context context) {
+      this.context = context;
+    }
+
+    @Override
+    public @NonNull LoadData<Bitmap> buildLoadData(@NonNull ConversationShortcutPhoto conversationShortcutPhoto, int width, int height, @NonNull Options options) {
+      return new LoadData<>(conversationShortcutPhoto, new Fetcher(context, conversationShortcutPhoto));
+    }
+
+    @Override
+    public boolean handles(@NonNull ConversationShortcutPhoto conversationShortcutPhoto) {
+      return true;
+    }
+
+    public static class Factory implements ModelLoaderFactory<ConversationShortcutPhoto, Bitmap> {
+
+      private final Context context;
+
+      public Factory(@NonNull Context context) {
+        this.context = context;
+      }
+
+      @Override
+      public @NonNull ModelLoader<ConversationShortcutPhoto, Bitmap> build(@NonNull MultiModelLoaderFactory multiFactory) {
+        return new Loader(context);
+      }
+
+      @Override
+      public void teardown() {
+      }
+    }
+  }
+
+  static final class Fetcher implements DataFetcher<Bitmap> {
+
+    private final Context                   context;
+    private final ConversationShortcutPhoto photo;
+
+    private Fetcher(@NonNull Context context, @NonNull ConversationShortcutPhoto photo) {
+      this.context = context;
+      this.photo   = photo;
+    }
+
+    @Override
+    public void loadData(@NonNull Priority priority, @NonNull DataCallback<? super Bitmap> callback) {
+      Bitmap bitmap;
+
+      try {
+        bitmap = getShortcutInfoBitmap(context);
+      } catch (ExecutionException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)